state-transition/core/state_processor.go

```diff
@@ -112,28 +123,30 @@ func (sp *StateProcessor) ProcessSlots(
 	if err != nil {
 		return nil, err
 	}
+	if slot == stateSlot {
+		return res, nil
+	}
+	if slot != stateSlot+1 {
+		return nil, fmt.Errorf("slot %d does not match expected slot %d", slot, stateSlot+1)
+	}
```

Comment:
somewhat a nit: after this enforcement that the passed in param `slot` must equal `stateSlot + 1`, would prefer using the passed in param `slot` below in this function.